base/common/python/pki/cli/main.py

```diff
@@ -175,35 +175,50 @@ def execute(self, argv):
 
             # get database path
             if args[i] == '-d':
-                self.database = args[i + 1]
+                try:
+                    self.database = args[i + 1]
+                except IndexError:
+                    break
```

Comment:
Should we generate an error message such as `Missing argument for option: d`? Otherwise the program would continue as if the parameter was not specified, then perform an incorrect operation.

Reply:
With this change, the code breaks out without parsing the rest of the arguments and then gets caught in the new `MissingArgumentException` catch block in the Java. The message reported to the terminal is e.g.:
```
pki -d
Missing argument for option: d
```
For the Python CLI you get two errors for a missing arg, one here as `IndexError` and then back in the Java as `MissingArgumentException`. I suppressed the Python exception and let the Java handle it so the error is the same if it is a Java CLI missing arg or a Python CLI missing arg.

Reply:
Just a fly-by question: why do this manually instead of using the argparse module?

Reply:
Presumably because the code was originally written in Python 2.x and `ArgParse` was added in Python 3.2.

Reply:
But yeah, `ArgParse` is very useful - I am a fan!

Reply:
I can't remember the reason why it's written like that, but it definitely needs an update.